bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ValuePair.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.internal.values;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Unordered value vector of size two, to hold two values being compared.
+ *
+ * @since 2201.9.0
+ */
+
+public class ValuePair {
+
+    Set<Object> valuePairSet = new HashSet<>(2);
+
+    public ValuePair(Object obj1, Object obj2) {
+        valuePairSet.add(obj1);
+        valuePairSet.add(obj2);
+    }
+
+    public boolean equals(Object o) {
+        if (!(o instanceof ValuePair valuePair)) {
```

Comment:
Do we need this code?

Reply:
It seems, there are no possibilities to this check being executed. Btw, IDEA gives warning and suggests this check. I think we can replace it with casting directly.